platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -78,6 +78,15 @@ void OfflineDatabase::initialize() {
     }
 }
 
+void OfflineDatabase::changePath(const std::string& path_) {
+    Log::Info(Event::Database, "Changing the database path.");
+
+    statements.clear();
```

Comment:
This is what we call on the destructor. This functions might throw and we should catch. To avoid duplicating the code, make both the destructor and this function call something like `OfflineDatabase::cleanup`